evm/src/main/java/org/hyperledger/besu/evm/code/CodeV0.java

```diff
@@ -39,24 +36,37 @@ public class CodeV0 implements Code {
   /** The bytes representing the code. */
   private final Bytes bytes;
 
+  private final int size;
+
   /** The hash of the code, needed for accessing metadata about the bytecode */
-  private final Supplier<Hash> codeHash;
+  private final Hash codeHash;
 
   /** Used to cache valid jump destinations. */
-  private long[] validJumpDestinations;
+  private long[] validJumpDestinationsBitMask;
 
   /** Code section info for the legacy code */
   private final CodeSection codeSectionZero;
 
   /**
    * Public constructor.
    *
-   * @param bytes The byte representation of the code.
+   * @param byteCode The byte representation of the code.
+   */
+  public CodeV0(final Bytes byteCode) {
+    this(byteCode, byteCode.isEmpty() ? Hash.EMPTY : Hash.hash(byteCode));
```

Comment:
I think it is better to keep the memoizer here on calculating the hash of the bytecode.